charabia/src/normalizer/lowercase.rs

```diff
@@ -9,10 +9,33 @@ use crate::Token;
 pub struct LowercaseNormalizer;
 
 impl Normalizer for LowercaseNormalizer {
-    // lowercasing characters doesn't change the characters length,
-    // so the `normalize` method is overloaded to skip the useless char_map computing.
+    // lowercasing characters cna change the characters length, so we need
+    // to make sure that the char mapping is correct and remap it if necessary.
+    // <https://github.com/meilisearch/charabia/pull/234>
     fn normalize<'o>(&self, mut token: Token<'o>, _options: &NormalizerOption) -> Token<'o> {
-        token.lemma = Cow::Owned(token.lemma().to_lowercase());
+        match token.char_map.take() {
+            Some(char_map) => {
+                // Most of the time lowercasing a string will not change it's length
+                let lowercased = token.lemma.to_lowercase();
+                if token.lemma.len() == lowercased.len() {
```

Comment:
Is it correct? One chunk of the string increases, and another may decrease to fake that nothing moved, but the chunk bounds moved, and bounds will be off. I fear that I'll need to remove this optimization...